clients/controller.go

```diff
@@ -0,0 +1,100 @@
+package clients
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/uber/aresdb/metastore/common"
+	"github.com/uber/aresdb/utils"
+	"io/ioutil"
+	"net/http"
+)
+
+// ControllerClient defines methods to communicate with ares-controller
+type ControllerClient interface {
+	GetSchemaHash(namespace string) (string, error)
+	GetAllSchema(namespace string) ([]common.Table, error)
+}
+
+// ControllerHTTPClient implements ControllerClient over http
+type ControllerHTTPClient struct {
+	c              *http.Client
+	controllerHost string
+	controllerPort int
+	headers        http.Header
+}
+
+// NewControllerHTTPClient returns new ControllerHTTPClient
+func NewControllerHTTPClient(controllerHost string, controllerPort int, heasers http.Header) *ControllerHTTPClient {
+	return &ControllerHTTPClient{
+		c:              &http.Client{},
+		controllerHost: controllerHost,
+		controllerPort: controllerPort,
+		headers:        heasers,
+	}
+}
+
+func (c *ControllerHTTPClient) GetSchemaHash(namespace string) (hash string, err error) {
+	var req *http.Request
+	req, err = c.getRequest(namespace, true)
+	if err != nil {
+		return
+	}
+	var resp *http.Response
+	resp, err = c.c.Do(req)
+	if err != nil {
+		return
+	}
+	if resp.StatusCode != http.StatusOK {
+		err = utils.StackError(fmt.Errorf("controller client error fetching hash"), fmt.Sprintf("code %d", resp.StatusCode))
+	}
+
+	var b []byte
+	b, err = ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return
+	}
+	hash = string(b)
+	return
+}
+
+func (c *ControllerHTTPClient) GetAllSchema(namespace string) (tables []common.Table, err error) {
+	var req *http.Request
+	req, err = c.getRequest(namespace, false)
+	if err != nil {
+		return
+	}
+	var resp *http.Response
+	resp, err = c.c.Do(req)
+	if err != nil {
+		return
+	}
+	if resp.StatusCode != http.StatusOK {
+		err = utils.StackError(fmt.Errorf("controller client error fetching schema"), fmt.Sprintf("code %d", resp.StatusCode))
```

Comment:
controller error fetching schema could be just in the error message

Reply:
what do you mean?

Reply:
"controller client error" could be in the message